pkg/providers/amifamily/ami.go

```diff
@@ -233,14 +276,18 @@ func getFiltersAndOwners(amiSelector map[string]string) ([]*ec2.Filter, []*strin
 	return filters, owners
 }
 
-func sortAMIsByCreationDate(amiRequirements map[AMI]scheduling.Requirements) []AMI {
-	amis := lo.Keys(amiRequirements)
-
+func sortAMIsByCreationDate(amis []AMI) []AMI {
 	sort.Slice(amis, func(i, j int) bool {
-		itime, _ := time.Parse(time.RFC3339, amis[i].CreationDate)
-		jtime, _ := time.Parse(time.RFC3339, amis[j].CreationDate)
-		return itime.Unix() >= jtime.Unix()
+		if amis[i].CreationDate != "" {
```

Comment:
Doesn't this need to check if `amis[j].CreationDate != ""` too?
```suggestion
		if amis[i].CreationDate != "" || amis[j].CreationDate != "" {
```